src/meta/processors/schemaMan/AlterTagProcessor.cpp

```diff
@@ -63,6 +63,23 @@ void AlterTagProcessor::process(const cpp2::AlterTagReq& req) {
         }
     }
 
+    if (existIndex) {
+        int64_t duration = 0;
+        if (prop.get_ttl_duration()) {
+            duration = *prop.get_ttl_duration();
+        }
+        std::string col;
+        if (prop.get_ttl_col()) {
+            col = *prop.get_ttl_col();
+        }
+        if (!col.empty() && duration > 0) {
```

Comment:
Can ttl and index coexist?

Reply:
Aha, hey can coexist. I saved the timestamp value to index val.

Reply:
`existIndex `  means there is an index,
`!col.empty() && duration> 0` means there is ttl

Why is there an error here?

Reply:
> `existIndex ` means there is an index,
> `!col.empty() && duration> 0` means there is ttl
> 
> Why is there an error here?

if index exists,  not allow to alter TTL properties.

Reply:
yeah, i know what you mean.
But the `prop ` schema prop is in the `schema`, not the schema prop in `AlterTagReq`.

Your current logic is that when alter tag, as long as there is index and ttl, an error will be reported.

Reply:
> yeah, i know what you mean.
> But the `prop ` schema prop is in the `schema`, not the schema prop in `AlterTagReq`.
> 
> Your current logic is that when alter tag, as long as there is index and ttl, an error will be reported.

yes，there should be an error thrown here, either `alter ttl ` or `alter column` (excepted for `add column` )

Reply:
I mean:
here logic is unreasonable.
As long as there are index and ttl,  here are not allowed to alter tag

Reply:
> I mean:
> here logic is unreasonable.
> As long as there are index and ttl, here are not allowed to alter tag

good point. 🐂B
![artimg](https://user-images.githubusercontent.com/56461666/114337268-6d32d100-9b83-11eb-8c5b-e5e141f5a933.gif)